src/awkward/contents/numpyarray.py

```diff
@@ -188,9 +188,14 @@ def _raw(self, nplike=None):
         return to_nplike(self.data, nplike, from_nplike=self._backend.nplike)
 
     def _form_with_key(self, getkey: Callable[[Content], str | None]) -> NumpyForm:
+        if hasattr(self._data, "inner_shape"):
+            inner_shape = self._data.inner_shape
+        else:
+            inner_shape = self._data.shape[1:]
+
         return self.form_cls(
             ak.types.numpytype.dtype_to_primitive(self._data.dtype),
-            self._data.shape[1:],
+            inner_shape,
```

Comment:
I thought that this had already been established in a recent PR... #3189.

I don't know why GitHub is showing it as a diff relative to `main`; it's already in `main`. Anyway, it's the right thing to do.